exe-unit/components/gsb-http-proxy/src/gsb_to_http.rs

```diff
@@ -51,11 +54,12 @@ impl GsbToHttpProxy {
             builder = headers::add(builder, message.headers);
 
             log::debug!("Calling {}", &url);
+            let response_monitor = requests_monitor.on_request().await;
             let response = builder.send().await;
             let response =
                 response.map_err(|e| GsbToHttpProxyError::ErrorInResponse(e.to_string()))?;
-
             let bytes = response.bytes().await.unwrap();
+            response_monitor.on_response();
```

Comment:
I did not want to drive request logic through some passing request function into `response_monitor`, and instead I just mark when to start billing and when to stop. If `on_response` is not called directly it will be on `Drop` (to make it work like lets say `Mutex`)